### PR TITLE
Added docs about deprecations

### PR DESCRIPTION
**Pull request by krichardsson**
Clarifications for APIs, versions and deprecations

Added dates to all deprecation commens to clarify when they will be removed

### CODE REVIEW

File: docs/development/apis_versions_deprecation.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+The purpose of managing APIs in a structured way is to help a developer to understand if a protocol or a set of
+functions is likely to change over time and if it will break the application. It should also be possible to detect if
+(and how) the API has changed and what changes that are required to the code to make the application work.
+
+## The update process and deprecations
+
+Areas that can be versioned using a version number, for instance protocols, should update the version number when
+something (of instance the protocol) is changed. Check the documentation for the specific area/protocol to see if there
+is a specific update policy for this area.
```
Comment (knmcguire) [inline docs/development/apis_versions_deprecation.md:23]: Could we perhaps mention the CRTP versioning here to make it more concrete?

File: docs/development/apis_versions_deprecation.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+being removed. Deprecated functions are documented in the release notes to give a heads up that they will be removed in
+a future release.
+
+## Current APIs with structured updates
+
+The following areas and protocols are updated with a structured process:
+
+* [CRTP](/docs/functional-areas/crtp)
+* App API - the functions called from `app_api/src/app_main.c`
+* [Parmeters](/docs/api/params.md) - parameters marked as `CORE`
```
Comment (knmcguire) [inline docs/development/apis_versions_deprecation.md:37]: typo: parameters

File: docs/development/apis_versions_deprecation.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+Areas that can be versioned using a version number, for instance protocols, should update the version number when
+something (of instance the protocol) is changed. Check the documentation for the specific area/protocol to see if there
+is a specific update policy for this area.
+
+For functions in the code base, there is not really a version number tied to the function (except the release version) that
+clearly tells the programmer if there has been an API break or not. The approach we use is to mark functions that we
+want to remove as 'deprecated`. The deprecated function will continue to exist and work for a minimum of 6 months before
+being removed. Deprecated functions are documented in the release notes to give a heads up that they will be removed in
+a future release.
```
Comment (knmcguire) [inline docs/development/apis_versions_deprecation.md:29]: Should we also mention that we should also add an alternative function suggestion to anything we deprecate? if one exists

File: src/modules/src/crtp_commander_high_level.c [hunk 4]
```diff
@@ -129,8 +129,8 @@ STATIC_MEM_TASK_ALLOC(crtpCommanderHighLevelTask, CMD_HIGH_LEVEL_TASK_STACKSIZE)
 // trajectory command (first byte of crtp packet)
 enum TrajectoryCommand_e {
   COMMAND_SET_GROUP_MASK          = 0,
-  COMMAND_TAKEOFF                 = 1, // Deprecated, use COMMAND_TAKEOFF_2
-  COMMAND_LAND                    = 2, // Deprecated, use COMMAND_LAND_2
+  COMMAND_TAKEOFF                 = 1, // Deprecated (removed after August 2023), use COMMAND_TAKEOFF_2
```
Comment (knmcguire) [inline src/modules/src/crtp_commander_high_level.c:132]: Wonder then what we should use for the command 1 and 2 then? And then we should also deprecate the functions that are connected to this as well

File: docs/development/apis_versions_deprecation.md (hunk 1 above)
Comment (krichardsson) [reply docs/development/apis_versions_deprecation.md]: Added

File: docs/development/apis_versions_deprecation.md (hunk 3 above)
Comment (krichardsson) [reply docs/development/apis_versions_deprecation.md]: Good point! Added

**Review by knmcguire — APPROVED** [review]

**Comment by knmcguire** [comment]
Looks good! I'll merge it.

**Pull request merged**